openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -0,0 +1,167 @@
+
+        # Test if value of these attributes match between the three objects
+        fields_to_test = [
+            'id',
+            'location',
+            'display_name',
+            'display_number_with_default',
+            'display_org_with_default',
+            'advertised_start',
+            'facebook_url',
+            'social_sharing_url',
+            'certificates_display_behavior',
+            'certificates_show_before_end',
+            'cert_name_short',
+            'cert_name_long',
+            'lowest_passing_grade',
+            'end_of_course_survey_url',
+            'mobile_available',
+            'visible_to_staff_only',
+            'location',
+            'number',
+            'url_name',
+            'display_name_with_default',
+            'start_date_is_still_default',
+            'pre_requisite_courses'
+        ]
+        for attribute_name in fields_to_test:
+            course_value = getattr(course, attribute_name)
+            cache_miss_value = getattr(course_overview_cache_miss, attribute_name)
+            cache_hit_value = getattr(course_overview_cache_hit, attribute_name)
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Test if return values all methods are equal between the three objects
+        methods_to_test = [
+            'clean_id',
+            'has_ended',
+            'has_started',
+            'start_datetime_text',
+            'end_datetime_text',
+            'may_certify',
+        ]
+        for method_name in methods_to_test:
+            course_value = getattr(course, method_name)()
+            cache_miss_value = getattr(course_overview_cache_miss, method_name)()
+            cache_hit_value = getattr(course_overview_cache_hit, method_name)()
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Other values to test
+        others_to_test = [(
+            course_image_url(course),
+            course_overview_cache_miss.course_image_url,
+            course_overview_cache_hit.course_image_url
+        ), (
+            get_active_web_certificate(course) is not None,
+            course_overview_cache_miss.has_active_web_certificates,
+            course_overview_cache_hit.has_active_web_certificates
+
+        )]
+        for (course_value, cache_miss_value, cache_hit_value) in others_to_test:
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+    @ddt.data(*itertools.product(
+        [
+            {
+                "static_asset_path": "/my/cool/path",
+                "display_name": "Test Course",
+                "start": YESTERDAY,
+                "end": TOMORROW,
+                "pre_requisite_courses": ['course-v1://edX+test1+run1', 'course-v1://edX+test2+run1']
+            },
```

Comment:
Also test with different values for `advertised_start`: with it being (1) a date string parse-able by fields.Date and (2) an arbitrary string (e.g., "Fall 2015").